File: Model/Export/Process/ExtendedObjectExportInfo.cs
using System.Collections.Generic;
using kCura.WinEDDS.Exporters;

namespace kCura.WinEDDS.Core.Model.Export.Process
{
	public class ExtendedObjectExportInfo : ObjectExportInfo
	{
		private readonly IFieldLookupService _fieldLookupService;

		public ExtendedObjectExportInfo(IFieldLookupService fieldLookupService)
		{
			_fieldLookupService = fieldLookupService;
		}
		public List<ViewFieldInfo> SelectedNativeFileNameViewFields { get; set; }

		public object GetFieldValue(string fieldName)
		{
			return Metadata[_fieldLookupService.GetOrdinalIndex(fieldName)];
		}

	}
}

File: Source/Relativity.Import.Export.Legacy/Exporters/ExtendedObjectExportInfo.vb
Imports System.Collections.Generic
Imports kCura.WinEDDS.Exporters

Namespace kCura.WinEDDS.FileNaming.CustomFileNaming
	Public Class ExtendedObjectExportInfo
		Inherits ObjectExportInfo

		Private ReadOnly _fieldLookupService As IFieldLookupService

		Public Sub New(fieldLookupService As IFieldLookupService)
			_fieldLookupService = fieldLookupService
			SelectedNativeFileNameViewFields = New List(Of ViewFieldInfo)()
		End Sub

		Public Property SelectedNativeFileNameViewFields As List(Of ViewFieldInfo)

		Public Function GetFieldValue(fieldName As String) As Object
			Return Metadata(_fieldLookupService.GetOrdinalIndex(fieldName))
		End Function

	End Class
End Namespace
